neuro_san/internals/run_context/langchain/core/langchain_run_context.py

```diff
@@ -564,14 +555,11 @@ async def wait_on_run(self, run: Run, journal: Journal = None) -> Run:
         if isinstance(verbose, str):
             verbose = bool(verbose.lower() in ("true", "extra", "logging"))
 
-        max_execution_seconds: float = agent_spec.get("max_execution_seconds",
-                                                      2.0 * MINUTES)
+        # max_execution_seconds: float = agent_spec.get("max_execution_seconds", 2.0 * MINUTES)
+
+        # Per advice from https://python.langchain.com/docs/how_to/migrate_agent/#max_iterations
         max_iterations: int = agent_spec.get("max_iterations", 20)
-        agent_executor = AgentExecutor(agent=self.agent,
-                                       tools=self.tools,
-                                       max_execution_time=max_execution_seconds,
-                                       max_iterations=max_iterations,
-                                       verbose=verbose)
+        recursion_limit: int = max_iterations * 2 + 1
```

Comment:
See comment above as to where this equation comes from.

Note we no longer use the concept of an AgentExecutor per the langchain 1.x constructs.
(Kinda too bad, I thought they were on the right conceptual track w/ this separation from Agent. 🤷 )